webpack-utils.ts

```diff
@@ -0,0 +1,89 @@
+// Copyright 2019 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ *
+ * @fileoverview Extra utilities used by webpack
+ */
+
+var PageMeta = function(name, description) {
+  this.name = name;
+  this.description = description;
+  this.referrer = 'no-referrer';
+  this.viewport = 'width=device-width, initial-scale=1.0, user-scalable=yes';
+
+  this['application-name'] = '<[ siteName ]>';
+  this['msapplication-square310x310logo'] = (
+    '<[ getAssetUrl(\'/assets/images/logo/msapplication-large.png\') ]>');
+  this['msapplication-wide310x150logo'] = (
+    '<[ getAssetUrl(\'/assets/images/logo/msapplication-wide.png\') ]>');
+  this['msapplication-square150x150logo'] = (
+    '<[ getAssetUrl(\'/assets/images/logo/msapplication-square.png\') ]>');
+  this['msapplication-square70x70logo'] = (
+    '<[ getAssetUrl(\'/assets/images/logo/msapplication-tiny.png\') ]>');
+
+  this['og:type'] = {
+    property: 'og:type',
+    content: 'article'
+  };
+  this['og:site_name'] = {
+    property: 'og:site_name',
+    content: 'Oppia'
+  };
+  this['og:url'] = {
+    property: 'og:url',
+    content: '<[pageUrl]>'
+  };
+  this['og:title'] = {
+    property: 'og:title',
+    content: name
+  };
+  this['og:description'] = {
+    property: 'og:description',
+    content: description
+  };
+  this['og:image'] = {
+    property: 'og:image',
+    content: '<[ getAssetUrl(\'/assets/images/logo/288x288_logo_mint.png\') ]>'
+  };
+};
+
+PageMeta.prototype.toHTML = function() {
```

Comment:
Interesting ... why are we doing all this? In general we very strongly prefer not constructing HTML like this in the JavaScript, as doing so introduces XSS risks. Ideally the HTML is written in HTML files.

Can you find a different way to do this? I note that https://github.com/jantimon/html-webpack-plugin#namecontent-meta-tags doesn't involve HTML construction via concatenation.

Reply:
The risk of XSS is mitigated by the fact that this is a **build** only script, int the production the pages will be already generatet with the `<meta>` tags.

But as I think of it @Jamesjay4199 can we insert the `<meta>` tags that are the same for all pages directly as an HTML into `header.html` and build dynamicly onle the ones that change for every page?

Reply:
Since we are using our own templates and loaders, https://github.com/jantimon/html-webpack-plugin does not add the meta tags automatically.

Yes @vojtechjelinek, that can be done. Meaning that only the title and description will be unique per page.

Reply:
Done @vojtechjelinek

Reply:
I would still want to remove the "HTML construction in JS" if possible, to avoid devs thinking that this sort of pattern is OK and copying it to use elsewhere. So, my question still stands -- can we find a different way to do this that doesn't involve building HTML manually via JS concatenation?

Reply:
Yes, I've done it a better way
PTAL